### PR TITLE
11511 create md filebacked

### PR DESCRIPTION
**Pull request by OwenArnold**
Original ticket is [here](http://trac.mantidproject.org/mantid/ticket/11511)

Have a mode whereby CreateMD can be run in-place. 
- Allow files to be passed as inputs as well as workspaces
- Add missing EFixed input parameter following Horace convention 
- Allow ConvertToMD to overwrite in-place
- Set the InPlace mode as an option. It won't be what people always want to do

Merging together existing MDWorkspaces as part of CreateMD will be handled separately 

**Tester**. Inspect unit tests and doc tests. This algorithm will be subjected to beta testing by scientists too.

### CODE REVIEW

**Comment by eXeC64** [comment]
You seem to be passing the new `EFix` property into the algorithm in the doc tests as `EFixed`. The documentation also refers to it as `EFixed`.

**Comment by DanNixon** [comment]
Is there a reason the parameter is not called ```EFixed``` as this is what this value is referred to in most other algorithms.

**Comment by OwenArnold** [comment]
@DanNixon The reason is to keep consistency with the Horace convention since CreateMD is a port of [gen_sqw](http://horace.isis.rl.ac.uk/Generating_SQW_files). If we are using `EFixed` consistently in other areas of Mantid, that would be a good case to adopt `EFixed` rather than continuing to use `EFix`

**Comment by DanNixon** [comment]
@OwenArnold That makes sense for Horace consistencey.

I suppose the best option would be some sort of parameter aliasing.

**Comment by martyngigg** [comment]
Conflicts with master have arisen.

File: Code/Mantid/docs/source/algorithms/CreateMD-v1.rst [hunk 1]
```diff
@@ -102,4 +105,42 @@ Output
...
+.. testoutput:: MultiConversionInPlace
+
+   [H,0,0]
+   [0,K,0]
+   [0,0,L]
+   DeltaE
+
+.. categories::
+
 .. categories::
```
Comment (martyngigg) [inline Code/Mantid/docs/source/algorithms/CreateMD-v1.rst]: I don't think we need both `.. categories::` lines. It produces 2 copies of the same line at the bottom of the page.

**Pull request merged**